### PR TITLE
Msw 91990846 test modal navigation

### PR DESCRIPTION
**Pull request by michaelwehrley**
Complete Navigation

* Alert services keeps track of navigation
* Refactor chapter and category and page classes to improve navigation
* Reorganize specs and refactor to some degree.
* Improve styling for Nexus 9.
* Capture completed categories in CurrentUser service.
* Capture the current category in the Current User service.
* Update chevron Font Awesome to right arrow.
* Add `CurrentUser` `initialize` specs.
* Update `Alert` functionality.
* Refactor constant usage.

### CODE REVIEW

**Comment by michaelwehrley** [comment]
I apologize b/c I moved some code around so it looks like more changes than it was.

**Comment by michaelwehrley** [comment]
My accident on the last commit - it was intend for a different branch.

File: test/unit/controllers/PagesControllerSpec.js [hunk 1]
```diff
@@ -162,4 +180,10 @@ describe('PagesController', function() {
       expect(controller.edNavConfig).to.equal(edNavConfig);
     });
   });
+
+  describe('pages.categoryConstants', function() {
```
Comment (ericcf) [inline test/unit/controllers/PagesControllerSpec.js]: Are tests such as this one valuable?

Comment (michaelwehrley) [inline test/unit/controllers/PagesControllerSpec.js]: Not the most important, but I wanted to easily verify that that injection was happening.  Cost benefit analysis? :smile:

File: test/unit/controllers/StartControllerSpec.js [hunk 2]
```diff
@@ -5,22 +5,19 @@ var expect = chai.expect;
...
+    controller = _$controller_;
     Payload = _Payload_;
   }));
 
-  describe('start.landingPath', function() {
+  describe('#landingPath', function() {
     it('sets the route to the landing page', function() {
-      var attributes, controller, path;
+      var attributes = Payload.where({ landing: true, type: 'page' })[0];
+      var path = '#/chapters/' + attributes.chapterId + '/pages/' + attributes.id;
```
Comment (ericcf) [inline test/unit/controllers/StartControllerSpec.js]: Does it concern you that the test and the implementation are identical?

File: test/unit/services/AlertSpec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  var Alert;
+
+  beforeEach(inject(function(_Alert_){
+    Alert = _Alert_;
+    Alert.start();
+  }));
+
+  describe('Alert.intialStage', function() {
+    it('returns 1', function() {
```
Comment (ericcf) [inline test/unit/services/AlertSpec.js]: ?

File: test/unit/services/AlertSpec.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    it('returns the next stage number', function() {
+      expect(Alert.currentStage).to.equal(0);
+
+      Alert.nextMessage();
+
+      expect(Alert.currentStage).to.equal(1);
+    });
+  });
+
+  describe('#previousMessage', function() {
```
Comment (ericcf) [inline test/unit/services/AlertSpec.js]: Is it worth testing edge cases? e.g. when `currentStage === 0`, what happens when you call `previousMessage()`?

**Comment by ericcf** [comment]
In general this looks fine, but I think it's worth considering each test and why you're writing it. Does it add net value? Does it test the right thing? etc.

File: test/unit/controllers/StartControllerSpec.js (hunk 2 above)
Comment (michaelwehrley) [inline test/unit/controllers/StartControllerSpec.js]: Yes.  That should be updated.
